src/gt4py/gtc/utils.py

```diff
@@ -0,0 +1,53 @@
+import collections.abc
+
+from devtools import debug  # noqa: F401
+
+
+def flatten_list(nested_iterables, filter_none=False, *, skip_types=(str, bytes)):
+    return list(flatten_list_iter(nested_iterables, filter_none, skip_types=skip_types))
+
+
+def flatten_list_iter(nested_iterables, filter_none=False, *, skip_types=(str, bytes)):
+    for item in nested_iterables:
+        if isinstance(item, list) and not isinstance(item, skip_types):
+            yield from flatten_list(item, filter_none)
+        else:
+            if item is not None or not filter_none:
+                yield item
+
+
+class ListTuple(collections.abc.Sequence):
```

Comment:
This thing does some non-obvious stuff and deserves a little bit of explanation, especially since it is in a general `utils` module. I would like to know where I should be using it and where not without hand parsing the code.

Reply:
done